cmake-format.yaml

```diff
@@ -0,0 +1,238 @@
+_help_parse: Options affecting listfile parsing
```

Comment:
[question] Do we actually change things from default here? Im starting to wonder about our policy of always including everything in config files like this as it makes it very hard to grok from cold what is actually salient about it.

Reply:
We do change things. Discussed with David and we've minimised it. Have made the change in main OpenAssetIO repor too.

https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/25/commits/5c99d7001c92d928c1a049e42946be363d490451